### DISCUSSION TITLE
How to add a public html file that is completely independent of react-router

### DISCUSSION THREAD

**Discussion by Matthematic** [remix-run/remix#6595] [Q&A]
Hello, I am new to react-router so this might be a simple question, but I am using create-react-app and I need to add a health check file to my app. This needs to be a standalone html file in the public folder that returns a 200 and can't be a "routed" page like the other pages in the app. My problem is that I can't get it setup so that this works, every time I go to the app at localhost:3000/health, react-router takes over and redirects to the 404 page since that isn't a handled route. Is there a simple way to do this, or am I confused?

**Comment by kiliman** [comment]
Which adapter are you using? Both Remix App Server and Express check files in the public folder before trying to send to Remix.

As for client navigations, you need to use regular `<a href="/health">` tags to bypass React Router. Or you can use `<Link to="/health" reloadDocument>` to force a full page load.

**Reply by kiliman** [reply]
Also, you may need to explicitly use `/health.html` (include extension) for files in the public folder. You would probably need to use the Express adapter and edit the _server.js_ file to configure the static handler.

```ts
app.use(express.static("public", { extensions: ['html'], maxAge: "1h" }));
```

https://expressjs.com/en/4x/api.html#express.static
